apis/core/v1beta1/openstackcontrolplane_types.go

```diff
@@ -520,6 +520,9 @@ func init() {
 
 // IsReady - returns true if service is ready to serve requests
 func (instance OpenStackControlPlane) IsReady() bool {
+	if instance.Status.Conditions == nil {
+		return false
+	}
```

Comment:
How is this function ever called before L144-148 of `controllers/core/openstackcontrolplane_controller.go` are called?  It looks like the only line that calls this func is L127 in ` controllers/core/openstackcontrolplane_controller.go`, and that deferred func would only be called after L144-148.

Reply:
the first time it enters `IsReady` the `InitConditions` isn't persisted yet (as it would only persist after the deferred function)
![image](https://github.com/openstack-k8s-operators/openstack-operator/assets/17153022/a3d57bbd-e148-432d-9247-244e7e6f9721)

Reply:
Yes, but it's still in memory in the object already, as per L144-148.  It doesn't need to be persisted to etcd yet.

Reply:
Wouldn't L145 call [1] before the `IsReady()` in the deferred func is called?

[1] https://github.com/openstack-k8s-operators/openstack-operator/blob/2f77a81b5d732149d77e0290eb88de929f7afc6f/apis/core/v1beta1/openstackcontrolplane_types.go#L535-L564

Reply:
it is called, but somehow it gets there empty

Reply:
Weird.  We should look into why that is, but this PR's approach works as a stop-gap for now.

Reply:
Can we have a test case that reproduce the problematic scenario? That would greatly help understanding the situation.

Reply:
Can it be the problem that InitConditions is run with a non-pointer receiver and therefore we loose the effect of instance.Status.Conditions.Init(&cl) ?